app/src/main/java/com/example/mystore/ui/profile/ProfileScreen.kt

```diff
@@ -123,12 +144,10 @@ fun ProfileScreen(
             SubmitButton(
                 enabled = true,
                 onClick = {
-                    viewModel.logout {
-                        destinationsNavigator.navigate(LoginScreenDestination)
-                    }
+                    viewModel.logout(onLogout = { navigateToLogin() })
```

Comment:
You should be able to pass it by reference: `viewModel.logout(onLogout = navigateToLogin)`. Anyway... merging!